### PR TITLE
feat: add colors to nixago logs

### LINKED ISSUE #47: Add helpful colors to nixago logs
Also, because nixago is a component tooling, make the log formatting configurable, defaulting to what `direnv` does as being a supposed primary integration target.

Also provide a log silencer flag, in case any integration tooling would prefer to choose an entirely different UX.

image

### PR DESCRIPTION
**Pull request by blaggacao**
closes: #47

### CODE REVIEW

**Comment by blaggacao** [comment]
That second commit still furthers style:

image

File: lib/hooks/ansi.nix [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Ansi escape codes
+let
+  # Nix strings only support \t, \r and \n as escape codes, so actually store
+  # the literal escape "ESC" code.
+  inherit (builtins) foldl' genList;
+  esc = "";
```
Comment (blaggacao) [inline lib/hooks/ansi.nix:6]: This should hold the literal escape code...

Looks like something in my auto ated pipeline has removed it. Should look like this: https://raw.githubusercontent.com/numtide/devshell/master/nix/ansi.nix

**Review by blaggacao — COMMENTED** [review]
~~Something went sideways while pushing...~~

All good: was a display issue on the GH cell phone app.

File: lib/hooks/ansi.nix (hunk 1 above)
Comment (blaggacao) [reply lib/hooks/ansi.nix:6]: No, all good, it's just that the GitHub app on the cell phone doesn't represent it right.

**Comment by jmgilman** [comment]
Thanks for this :) I'll give it a look through later today and try to get it merged.

**Comment by jmgilman** [comment]
LGTM. Thanks for the PR!

**Pull request merged**